src/renderer/helpers/subscriptions.js

```diff
@@ -140,3 +140,36 @@ export function addPublishedDatesInvidious(videos) {
     return video
   })
 }
+
+export async function loadSubscriptionVideosFromCacheOrServer(subscriptionComponent) {
+  subscriptionComponent.isLoading = true
```

Comment:
I don't think this is a good practice to
- pass the component inside
- assume the properties of component

This seems like a parent component's job (or mixin?)

Reply:
Ask @absidue for more~

Reply:
Mixins aren't a best practice, and neither is this I imagine. But I'm not sure what's a better practice in Vue 2 for handling code duplication.

Reply:
A [renderless component](https://vuejs.org/guide/components/slots.html#renderless-components) as a parent, maybe? Not sure, but I can look into it if this is a serious enough code quality concern. Will wait to hear back first, though.

Reply:
I agree with Pikachu here, passing in a component seems like such a bad idea, especially as this looks like it could be restructured to avoid that without too much work.

1. Don't set loading in here, that can be done in the component before and after the function call.
2. Use function parameters to pass the values you need in, e.g. fetchSubscriptionsAutomatically or just access that directly from the store in here.
3. You are allowed to return values from functions, in this case you can do it instead if setting the property on the component.

Reply:
I get that that's better practice, but it is annoying that all of the glue code requires almost identical code in each file. It ends up not really reducing the code duplication in these files much. Is what was in [this commit](https://github.com/jasonhenriquez/FreeTube/pull/2/commits/b89043e26896a5f27eeb66221b38090298778776) closer to what you're getting at?